airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -355,6 +358,17 @@ private static void assertColumnsWithSameNameAreSame(String schemaName, String t
                 })
                 .collect(Collectors.toList())))
         .collect(Collectors.toList());
+    result.forEach(t -> {
+      try {
+        final List<String> primaryKeys = database.bufferedResultSetQuery(
+            conn -> conn.getMetaData().getPrimaryKeys(databaseOptional.orElse(null), t.getSchemaName(), t.getName()),
+            resultSet -> resultSet.getString(JDBC_COLUMN_COLUMN_NAME));
+        t.addPrimaryKeys(primaryKeys);
```

Comment:
One convention that we try to follow in our java code is to keep objects immutable wherever possible. This pattern of setting fields after initialization should only be used if absolutely necessary (e.g. it represents different stages of the object's lifecycle). You'll notice that in almost all of our java classes all of the fields are final. Here it is not necessary to set the field late. The integrity of this object is compromised by because we happen to use two separate queries to construct it.

I would suggest the following you do both queries 1. get all columns. 2 get all primary keys. and then construct the table info one time (no need for addPrimaryKeys).

LMK if any of this is unclear or if the motivation doesn't make sense.